src/components/ExportModal/ExportModal.js

```diff
@@ -0,0 +1,188 @@
+  componentWillMount() {
+    this.props.loadEntities({
+      href: '/bookmarks?per_page=100',
+      type: ARTICLES_VIEW_STATE,
+      schema: arrayOf(articleSchema),
+    }).then(() => {
+      this.loadAllArticlesFromServer(this.props.linkHeader.get('last'));
+    });
+  }
+
+  onRowSelection(selectedRows) {
+    this.setState({ selectedRows });
+  }
+
+  loadAllArticlesFromServer(last) {
+    if (!last) {
+      return;
+    }
+
+    const pageLast = parseInt(last.get('page'), 10);
+    const perPageLast = parseInt(last.get('per_page'), 10);
+    const total = pageLast * perPageLast;
+    const perPage = Math.trunc(Math.min(MAX_PER_PAGE, total) / perPageLast) * perPageLast;
+    const numberOfRequests = Math.ceil(total / perPage);
+
+    [...Array(numberOfRequests).keys()].reduce((seq, i) => {
+      const page = i + ONE;
+
+      return seq.then(() => this.props.loadEntities({
+        href: `/bookmarks?per_page=${perPage}&page=${page}`,
+        type: ARTICLES_VIEW_STATE,
+        schema: arrayOf(articleSchema),
+        resetState: page === ONE,
+      }));
+    }, Promise.resolve());
+  }
+
+  bind() {
+    this.onRowSelection = this.onRowSelection.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit() {
+    const tempLink = document.createElement('a');
+    const newArticles = getSelectedArticles(this.props.articles, this.state.selectedRows);
+    const content = encodeURIComponent(JSON.stringify(newArticles, null, TWO));
+
+    if (newArticles.isEmpty()) {
+      return;
+    }
+
+    tempLink.setAttribute('href', `data:text/plain;charset=utf-8,${content}`);
+    tempLink.setAttribute('download', 'heutagogy.json');
+
+    if (document.createEvent) {
+      const event = document.createEvent('MouseEvents');
+
+      event.initEvent('click', true, true);
+      tempLink.dispatchEvent(event);
+    } else {
+      tempLink.click();
+    }
+  }
+
+  handleClose() {
+    this.props.unmount();
+  }
+
+  renderStatus() {
+    return (
+      <div>
+        { this.props.loadingArticlesStatus && this.props.loadingArticlesStatus.get('isInProgress')
+          ? <div className={styles.linearProgress}>
+            <div style={{ margin: 10 }}>{'Loading articles...'}</div>
+            <LinearProgress mode="indeterminate" />
+          </div> : null }
+        { this.props.loadingArticlesStatus && this.props.loadingArticlesStatus.get('isFailed')
+          ? <div className={styles.errorMessage}><i>{this.props.loadingArticlesStatus.get('message')}</i></div> : null }
+      </div>
+    );
+  }
+
+  render() {
+    const actions = [
+      <FlatButton
+        disabled={this.props.articles.isEmpty()}
+        label={'Save'}
+        primary
+        style={inlineStyles.submit}
+        onTouchTap={this.handleSubmit}
+      />,
+      <FlatButton
+        label={'Cancel'}
+        primary
+        style={inlineStyles.close}
+        onTouchTap={this.handleClose}
```

Comment:
(of course I can rename `unmount` to `handleUnmount` but it's really nitpicking)